packages/traceability/issuer-api/src/pods/certificate/transaction-log.entity.ts

```diff
@@ -0,0 +1,23 @@
+import { Column, Entity, PrimaryGeneratedColumn, Index } from 'typeorm';
+import { BlockchainEventType } from './types';
+
+export const TRANSACTION_LOG_TABLE_NAME = 'transaction_log';
+
+@Entity({ name: TRANSACTION_LOG_TABLE_NAME })
+export class TransactionLog {
+    @PrimaryGeneratedColumn()
+    id: number;
+
+    @Index()
+    @Column({ type: 'int4' })
```

Comment:
Use `bigint` instead of `int4`, since the certificate ID can be up to `2^256 - 1`.

Reply:
This also needs to be changed in other parts of the codebase, but let's leave it for another time.

Reply:
We should probably also add a check `Min(1)`.

Reply:
What's that check? Is it from TypeOrm? I cannot find this

Reply:
`class-validator` package

Reply:
Like in https://github.com/energywebfoundation/origin/blob/0cdd344dbc718d5048457c95d4d2dada4d45ad96/packages/traceability/issuer-api/src/pods/certificate/certificate.entity.ts#L13

Reply:
But why to use class-validator here? It's not validated anywhere (transaction log cannot be created as a input from API or something where it would be validated by Nest), and the object before creation is not validated by class-validator manually anywhere.

Afaik adding this decorator changes nothing, if we don't explicitly or implicitly validate this with class-validator functions.

Could you shed more light on this?

Reply:
If we add class-validator in the entity, it will be validated on `repository.save()`. Indeed, we'll almost 100% always get a valid certificate ID from an event, but still it's an additional layer of validation.

I'm also fine if you choose to leave this out.

Reply:
https://orkhan.gitbook.io/typeorm/docs/validation

Reply:
Seems like you might need to manually call `validate()` before `.save()`.

Reply:
I can do this I think

Reply:
Added! :)